### ISSUE TITLE
ops(compute-window): preserve checkpoint bytes, model lineage, and loadability before host loss

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#8831]
## Archetype Metadata

```yaml
archetype: artifact-preservation-tooling
resource: local
execution_mode: unattended
one_pr: true
parallel_safe: true
evidence_tier: operational
linked_policy:
  - docs/context/artifact_evidence_vocabulary.md
  - docs/maintainer_values.md
```

## Parent

Child of the compute-window preservation programme.

## Goal / Problem

Checkpoints and companion model files can remain referenced by configs while the only usable bytes, optimizer state, normalization data, or environment metadata remain on a host that will soon be inaccessible. Add a fail-closed preservation workflow for model artifacts already owned by current registries and campaign/training manifests.

## Scope

- Inventory checkpoint/model references from active compute-window workloads, current registries, configs, training manifests, and artifact receipts.
- Resolve only through existing model/artifact owners.
- Record artifact ID, producer source/config/data identities, framework/version, observation/action contract, architecture, normalization, seed, training status, expected companion files, byte size, SHA-256, storage class, current availability, and downstream consumers.
- Verify loadability in metadata-only or no-inference mode when the framework owner supports it.
- Preserve exact bytes to a durable destination through the approved transfer workflow.
- Reject missing companions, ambiguous aliases, mutable `latest` references, digest mismatch, incompatible framework, incomplete training output, private-only references, and rights-blocked publication assumptions.
- Generate a sanitized public manifest and private custody receipt.
- Add fixtures for complete SB3/PyTorch-like checkpoint sets, missing normalizer, stale alias, wrong observation contract, digest drift, partial copy, and loadability unavailable.

## Out of Scope

- training or evaluating models;
- changing model registries, checkpoint formats, architectures, or observation contracts;
- publishing model bytes or asserting redistribution rights;
- repairing corrupted checkpoints;
- treating metadata loadability as planner-performance evidence.

## Affected Files

One preservation checker/manifest generator or bounded extension, focused tests, and concise model-custody documentation. Maximum 12 files and 900 net new lines.

## Definition of Done

- [ ] Every active workload checkpoint reference has one exact availability and custody status.
- [ ] Required companion files and contracts are explicit.
- [ ] Mutable aliases cannot pass as durable identity.
- [ ] Preserved bytes reproduce source checksums at the destination.
- [ ] Loadability status is distinct from performance or evidence status.
- [ ] Public output contains no private paths or unauthorized bytes.
- [ ] Unknown rights remain unknown and do not block private preservation.

## Validation / Testing

```bash
uv run pytest -q <focused checkpoint preservation tests>
uv run python <checkpoint-preservation-tool> --check --fixture <checkpoint-fixture> --format json
uv run ruff check <changed Python files>
uv run ruff format --check <changed Python files>
git diff --check
PR_READY_MODE=final BASE_REF=origin/main scripts/dev/pr_ready_check.sh
```

## Stop Conditions

Stop on ambiguous artifact identity, missing producer lineage, incomplete required companions, digest mismatch, unsafe destination, or a proposal to publish uncleared bytes.

## Project Metadata

```yaml
priority: 0
expected_duration_hours: 7
requires_domain_review: false
requires_compute_authority: false
requires_external_input: false
```

## Blocked by

None. Ready for unattended local implementation and fixture validation.
